app/src/main/java/mozilla/lockbox/store/AutoLockStore.kt

```diff
@@ -0,0 +1,68 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package mozilla.lockbox.store
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.os.SystemClock
+import android.preference.PreferenceManager
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.rxkotlin.addTo
+import mozilla.lockbox.action.LifecycleAction
+import mozilla.lockbox.action.Setting
+import mozilla.lockbox.flux.Dispatcher
+import mozilla.lockbox.support.Constant
+
+class AutoLockStore(
+    val dispatcher: Dispatcher = Dispatcher.shared,
+    val settingStore: SettingStore = SettingStore.shared,
+    val lifecycleStore: LifecycleStore = LifecycleStore.shared
+) : ContextStore {
+
+    private val compositeDisposable = CompositeDisposable()
+    private lateinit var preferences: SharedPreferences
+
+    init {
+        lifecycleStore.lifecycleFilter
+            .filter { it == LifecycleAction.Foreground }
+            .switchMap { settingStore.autoLockTime.take(1) }
+            .subscribe(this::checkElapsedTime)
+            .addTo(compositeDisposable)
+
+        lifecycleStore.lifecycleFilter
+            .filter { it == LifecycleAction.Background }
+            .map { SystemClock.elapsedRealtime() }
+            .subscribe {
+                preferences
+                    .edit()
+                    .putLong(Constant.Key.backgroundedTime, it)
+                    .apply()
+            }
+            .addTo(compositeDisposable)
+    }
+
+    override fun injectContext(context: Context) {
+        preferences = PreferenceManager.getDefaultSharedPreferences(context)
+    }
+
+    private fun checkElapsedTime(autoLockTime: Setting.AutoLockTime) {
+        val lastBackgroundTime = preferences.getLong(Constant.Key.backgroundedTime, -1)
+        val currentSystemTime = SystemClock.elapsedRealtime()
+
+        val elapsedMS = currentSystemTime - lastBackgroundTime
+
+        when {
+            // edge case: someone has backgrounded the app after the device has been running for x minutes.
```

Comment:
@jhugman would appreciate thoughts on strategy here -- should the store dispatch the lock / unlock actions?

in iOS we basically have the datastore read the stored `backgroundedTime` equivalent and decide whether it is locked or unlocked IIRC, but it would be nice to have it a more contained to this `AutoLockStore` class.